TYPO3.Media/Classes/TYPO3/Media/Controller/AssetController.php

```diff
@@ -175,8 +175,13 @@ public function indexAction($view = null, $sort = null, $filter = null, $tagMode
             }
         }
 
-        if ($this->browserState->get('sort') !== 'Modified') {
-            $this->assetRepository->setDefaultOrderings(array('resource.filename' => \TYPO3\Flow\Persistence\QueryInterface::ORDER_ASCENDING));
+        if ($this->browserState->get('sort') !== 'Modified-ASC') {
+            $sortOptions = explode('-', $this->browserState->get('sort'));
+
+            ($sortOptions[0] == 'Name') ? $sortFor = 'resource.filename' : $sortFor = 'lastModified';
```

Comment:
not sure if it's technically disallowed in the CGL, but it's surely an unconventional way of writing such a statement which I don't think exists elsewhere, so should be written as ``$sortFor = $sortOptions[0] == 'Name' ? 'resource.filename' : 'lastModified';`` IMO.. you can keep the ``()`` if you prefer